--- phpBB2/common.php	2008-02-10 18:19:56.000000000 +0100
+++ patched-common.php	2019-04-26 12:38:24.433751882 +0200
@@ -193,16 +193,20 @@
 unset($dbpasswd);
 
 //
 // Obtain and encode users IP
 //
-// I'm removing HTTP_X_FORWARDED_FOR ... this may well cause other problems such as
-// private range IP's appearing instead of the guilty routable IP, tough, don't
-// even bother complaining ... go scream and shout at the idiots out there who feel
-// "clever" is doing harm rather than good ... karma is a great thing ... :)
-//
-$client_ip = ( !empty($HTTP_SERVER_VARS['REMOTE_ADDR']) ) ? $HTTP_SERVER_VARS['REMOTE_ADDR'] : ( ( !empty($HTTP_ENV_VARS['REMOTE_ADDR']) ) ? $HTTP_ENV_VARS['REMOTE_ADDR'] : getenv('REMOTE_ADDR') );
+$forwarded_index = getenv("PHPBB_X_FORWARDED_FOR_INDEX");
+if ($forwarded_index !== false && isset($_SERVER['HTTP_X_FORWARDED_FOR']))
+{
+    $forwarded_for = explode(',', $_SERVER['HTTP_X_FORWARDED_FOR']);
+    $client_ip = trim($forwarded_for[count($forwarded_for) + (int)$forwarded_index]);
+}
+else
+{
+    $client_ip = ( !empty($HTTP_SERVER_VARS['REMOTE_ADDR']) ) ? $HTTP_SERVER_VARS['REMOTE_ADDR'] : ( ( !empty($HTTP_ENV_VARS['REMOTE_ADDR']) ) ? $HTTP_ENV_VARS['REMOTE_ADDR'] : getenv('REMOTE_ADDR') );
+}
 $user_ip = encode_ip($client_ip);
 
 //
 // Setup forum wide options, if this fails
 // then we output a CRITICAL_ERROR since
